expression/builtin_arithmetic.go

```diff
@@ -180,6 +180,11 @@ type builtinArithmeticPlusIntSig struct {
 	baseBuiltinFunc
 }
 
+func (s *builtinArithmeticPlusIntSig) Clone() builtinFunc {
+	newSig := &builtinArithmeticPlusIntSig{baseBuiltinFunc: *s.baseBuiltinFunc.clone()}
```

Comment:
How about:
```
newSig := builtinArithmeticPlusIntSig{}
newSig.cloneFrom(s.baseBuiltinFunc)

func (b *baseBuiltinFunc) cloneFrom(from *baseBuiltinFunc) {
	args =   make([]Expression, 0, len(from.args))
	for _, arg := range from.args {
		args = append(args, arg.Clone())
	}
	b.args = args
	b.ctx  =   from.ctx,
	b.tp =     from.tp,
	b.pbCode = from.pbCode,
}
```

We can reduce many allocation this way.

Reply:
I'll have a try.